vignettes/piggyback.Rmd

```diff
@@ -0,0 +1,283 @@
+---
+title: "Piggyback Data atop your GitHub Repository!"
+author: "Carl Boettiger & Tan Ho"
+date: "2023-12-26"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{piggyback}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>",
+  results="hide",
+  eval = Sys.getenv("TAN_GH_TOKEN", FALSE)
+)
+
+Sys.setenv(piggyback_cache_duration=0)
+```
+
+## Why `piggyback`?
+
+`piggyback` grew out of the needs of students both in my classroom and in my research 
+group, who frequently need to work with data files somewhat larger than one can 
+conveniently manage by committing directly to GitHub.  As we frequently want to 
+share and run code that depends on >50MB data files on each of our own machines, 
+on continuous integration, and on larger computational servers, data sharing 
+quickly becomes a bottleneck. 
+
+[GitHub allows](https://docs.github.com/en/github/managing-large-files/distributing-large-binaries)
+repositories to attach files of up to 2 GB each to releases as a way to distribute 
+large files associated with the project source code.  There is no limit on the 
+number of files or bandwidth to deliver them.  
+
+## Authentication
```

Comment:
Rewrote authentication to point to usethis guide as best practice, and envvar as backup/noninteractive option.

Reply:
oh nice this is here.  I think this can be mentioned briefly in README though too

Reply:
Added. Now reads: 

> Downloading from private repos or uploading to any repo requires authentication, specifically a GitHub Personal Access Token (PAT). This can be stored as a [gh::gh_token()](https://usethis.r-lib.org/articles/git-credentials.html#get-a-personal-access-token-pat) or a GITHUB_PAT environment variable - for more information, see the vignette notes on [authentication](https://docs.ropensci.org/piggyback/articles/piggyback.html#authentication).